app/code/Magento/Swatches/Model/Plugin/EavAttribute.php

```diff
@@ -209,15 +218,14 @@ protected function processSwatchOptions(Attribute $attribute)
      * Get options array without deleted items
      *
      * @param array $optionsArray
+     *
      * @return array
      */
     protected function prepareOptionIds(array $optionsArray)
     {
         if (isset($optionsArray['value']) && is_array($optionsArray['value'])) {
             foreach (array_keys($optionsArray['value']) as $optionId) {
-                if (isset($optionsArray['delete']) && isset($optionsArray['delete'][$optionId])
-                    && $optionsArray['delete'][$optionId] == 1
-                ) {
+                if (isset($optionsArray['delete'][$optionId]) && $optionsArray['delete'][$optionId] == 1) {
```

Comment:
Could you please simplify this nested ifs, handling the negative case first and use a strict comparison operator?

Reply:
Hi @Stepa4man, done